examples/new_api_usage.jl

```diff
@@ -0,0 +1,20 @@
+# for features with built-in data, there will be convenience constructors that can build AtomFeats with default binning, etc. just from the names, so you could do this...
+feature_names = ["Group", "Row", "Block", "Atomic mass", "Atomic radius", "X"]
+features = AtomFeat.(feature_names)
+
+# or, of course, have a higher degree of control by specifying bin numbers, etc. and then just doing something like...
+features = AtomFeat.(zip(feature_names, num_bins, logspaced))
+
+# the point is, rarely should the user ever have to write the encode_f and decode_f themselves (really only if they're defining their own custom features)
```

Comment:
Do we need to shy away from having users write `encode`? Ideally, I would embrace it and just have a bunch defined already so in practice you never have to, but if you need it its easy and there is precedent already on how to handle various scenarios without using `global` for ex.

Reply:
Yeah that's basically what I'm envisioning, maybe my explanation wasn't clear. When I say they should rarely have to write it, I mean if they're only using built-in features they should never have to, and then if they want to add some custom features then they probably will. Though I'd like to have some way to automatically generate them for the simple case where the new features are just additions to the lookup table.

Reply:
Thoughts on making it dispatch on the abstract feature type and have that be a lookup? Users can overload it for more precise definitions of more specific feature types.

Reply:
I'm not sure it really makes sense to "preference" noncontextual features in that way...

Reply:
Alright